### ISSUE TITLE
Completion not available on PHP class inside project

### ISSUE THREAD

**Issue by LukeMauldin** [m2mdas/phpcomplete-extended-laravel#1]
I have a Laravel project with the structure below.  If I open server.php, the plugin detects the composer.json file and generates the index correctly.  However, if I open BaseController.php the plugin does not correctly detect that BaseController.php is part of a Laravel project and does not provide code completion.

-{app}
  -server.php
  -composer.json
  -src
    -controllers
       -BaseController.php

**Comment by m2mdas** [comment]
I just checked locally, completion is working here. Can you run `ls -lsh .phpcomplete_extended/` command in your project directory and paste the output? There is file named `report.txt` in that folder that has the list of the files not indexed. Also autocompletion works if vim `:pwd` is project root directory.

**Comment by LukeMauldin** [comment]
I did not realize my current directory needed to remain at the project root when I opened the new files.  I didn't see that documented anywhere?  It would be nice if somehow the plugin could be smart enough to search in parent directories to see if an index or composer.json file existed.

**Comment by m2mdas** [comment]
Yes, I agree that it should be documented, I will document about it tomorrow. 

I have thought about intelligently detecting project. There are other plugins such as tpope's `fugitive`, shougu's `unite.vim` do that. But in this plugins case I detect a project is a composer project or not by checking the existence of `composer.json` in the directory. But the vendor dependency projects also have `composer.json`, so whenever you go to a vendor dependency files during code browsing the plugin have two options, set closest `composer.json` dir to project dir which is not right or check to defined depth and prompt you to select form one of the directories that have `composer.json` which can be irritating if you move around dependencies more. Thats why I did not implement it. Besides in the era of `CtrlP` and `Unite.vim` it is not much of a problem to switch/open a file/buffer. I have also added `phpcomplete/files` `unite.vim` source the lists all the valid autoloaded classes. So I think it will not be much of problem :)
